ckanext/saml2/plugin.py

```diff
@@ -239,6 +275,10 @@ def identify(self):
             log.info("Couldn't decode nameid, giving up")
             return
 
+        user_info = saml2_get_user_info(c.user).first()
+        if user_info is not None:
+            c.user = user_info.user_name
```

Comment:
Following on from my comment on the Jira ticket. The pattern used on this line causes problems when the username is changed.  It should take the user id from the saml2_user table and subsequently get the current username from the user table, alternatively `saml2_get_user_info()` could be modified to join user and saml2_user tables and return the complete and accurate user info.